script.js

```diff
@@ -0,0 +1,116 @@
+//interactive navigation menu
+document.addEventListener('scroll', onScroll);
+
+function onScroll(event) {
+  const curPos = window.scrollY;
+  const ancher_block = document.querySelectorAll('#body>div, section');
+  const links = document.querySelectorAll('#menu a');
+  const linksBurger = document.querySelectorAll('#menu-burger a');
+  ancher_block.forEach((el) => {
+    if (el.offsetTop <= curPos && (el.offsetTop + el.offsetHeight) > curPos) {
+      links.forEach((a) => {
+        a.classList.remove('active');
+        if (el.getAttribute('id') === a.getAttribute('href').substring(1)) {
+          a.classList.add('active');
+        }
+      })
+    }
+  });
+  ancher_block.forEach((el) => {
+    if (el.offsetTop <= curPos && (el.offsetTop + el.offsetHeight) > curPos) {
+      linksBurger.forEach((a) => {
+        a.classList.remove('active');
+        if (el.getAttribute('id') === a.getAttribute('href').substring(1)) {
+          a.classList.add('active');
+        }
+      })
+    }
+  });
+}
+
+// interactive tab menu
+const TAB = document.getElementById('tab');
+
+TAB.addEventListener('click', (event) => {
+  TAB.querySelectorAll('span').forEach(el => el.classList.remove('tag_selected'));
+  event.target.classList.add('tag_selected');
+  shuffleNodes();
+});
+
+//shuffle tab menu
+let list = document.getElementById("list");
+
+function shuffle(items) {
+  let cached = items.slice(0), temp, i = cached.length, rand;
+    while(--i) {
+      rand = Math.floor(i * Math.random());
+      temp = cached[rand];
+      cached[rand] = cached[i];
+      cached[i] = temp;
+    }
+    return cached;
+};
+
+function shuffleNodes() {
+  let nodes = list.children, i = 0;
+  nodes = Array.prototype.slice.call(nodes);
```

Comment:
What is going on in this line? You redeclare nodes using `Array.prototype.slice` (why not just `nodes.slice()`, for example ?).

Reply:
Because I am working with an array of data!